pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -749,13 +750,35 @@ func (s *Service) DiscoverLaunchTemplateAMI(scope scope.LaunchTemplateScope) (*s
 		imageLookupBaseOS = scope.GetEC2Scope().ImageLookupBaseOS()
 	}
 
+	instanceType := lt.InstanceType
+	// If instance type is not specified on a launch template, we can safely assume the instance type will be a `t3.medium`.
```

Comment:
Since this should be input from users, we could fix this in the E2E test templates/cluster configs rather than hardcoding here.

Reply:
I'll move my comment over here as it'll be easier:

When running the tests with a launch template, the instanceType field is left empty, this is because AWS automatically populates this with t3.medium on their end when using a launch template.

I've added a check here to just assume the instanceType is a t3.medium in this case.

Obviously this is hardcoded and AWS could change this value at some point, do you think instead of assuming the instance type, we should just default the architecture x86_64 if the instance type is empty instead?

Reply:
> this is because AWS automatically populates this with t3.medium on their end when using a launch template.

Is this always set to m3.medium? Is it not taken as inout from user in launch template?

Reply:
According to AWS it's always set to `t3.medium`, unless a user specifies an instance type in their launch template, then it's overridden. I'll push a change to default the architecture instead as that was the previous behavior